### PR TITLE
Regenerate APIs with required spec, optional status.atProvider

### LINKED ISSUE #697: All provider-aws CRDs should be conformant
### What happened?

We'd like to certify provider-aws as a [conformant Crossplane provider](https://github.com/cncf/crossplane-conformance). My first pass at running the conformance tests failed due to some non-conformant CRD schemas, which you can see at https://github.com/cncf/crossplane-conformance/issues/4. Specifically it looks like:

* A whole bunch of resources mark `status.atProvider` as a required field. We prefer not to mark any status fields as "required", because we don't expect humans (or any entity other than the managed resource controller) to write those statuses.
* Several managed resources don't require a `spec`. I imagine there are some edge cases where a `spec` really is optional, but for the sake of being able to test that the broad majority of resources are doing the right thing I think we should keep requiring `spec`. I believe if `spec` is required but none of the `spec` fields are required it's possible to write `spec: {}` to indicate this.

### How can we reproduce it?

Follow the instructions at https://github.com/cncf/crossplane-conformance/blob/main/instructions.md for Crossplane 1.2 and provider-aws v0.18.1.

### What environment did it happen in?
Crossplane version: 1.2
Provider version: 0.18.1

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

Fixes #697
Depends on aws-controllers-k8s/code-generator#81

In the vast majority of resources a spec is required, so I'd like to mark it as such. I believe that in the edge cases where the spec really contains only optional fields it's possible to write spec: {}. Note that technically making `spec` required is a breaking change.

We prefer not to mark status fields as required, since resources will always exist at some point without their status and we consider 'requireness' to be more of a human-facing constraint than a software-facing constraint that our managed resource controllers should be subject to.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

I don't imagine the `atProvider` change will affect anyone. I'd like to assess how many (if any) of the affected resources _really_ could have had an empty spec and still functioned. If any exist, I'd like to confirm my belief that folks can write `spec: {}` to indicate that their managed resource doesn't need a spec.

### CODE REVIEW

**Review by negz — COMMENTED** [review]
> I'd like to assess how many (if any) of the affected resources really could have had an empty spec and still functioned.

It appears that all generated resources require `spec.forProvider.region`, which implies that none of these would have worked with an empty spec. I think in practice that means this change will not be breaking for anyone.

File: Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ PROJECT_REPO := github.com/crossplane/$(PROJECT_NAME)
 
 PLATFORMS ?= linux_amd64 linux_arm64
 
-CODE_GENERATOR_COMMIT ?= v0.2.1
+CODE_GENERATOR_COMMIT ?= c4592885b843b2f0a316718a15349ea07b3a50b1
```
Comment (negz) [inline Makefile:9]: How do we feel about pinning to a commit rather than a specific release? It looks we just missed out on a code-generator release yesterday.

Comment (muvaf) [reply Makefile:9]: I think that's fine.

File: Makefile [hunk 2]
```diff
@@ -79,16 +79,6 @@ crds.clean:
 
 generate.run: go.generate crds.clean
```
Comment (muvaf) [inline Makefile:81]:
```suggestion
generate.run: services.all go.generate crds.clean

```

We intentionally override `reviewable` here to make sure generated code is up to date. Maybe add `services.all` to `generate.init` or `generate.run`?

**Review by muvaf — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (negz) [reply Makefile:81]: Ah right - will fix.

Comment (negz) [reply Makefile:81]: I took a stab at this, but it's not quite working how I'd like. For whatever reason `go generate` runs _before_ `services.all` - rather than after possibly because it's part of the Go make libraries?

Comment (negz) [reply Makefile]: Found a way to fix this.

File: Makefile (hunk 1 above)
Comment (A-Hilaly) [reply Makefile:9]: `v0.2.2` has been released and includes the recent crossplane changes :) https://github.com/aws-controllers-k8s/code-generator/releases/tag/v0.2.2

**Review by muvaf — APPROVED** [review]
Thanks @negz !
